demos/plots/quadrangle_mesh.py

```diff
@@ -0,0 +1,60 @@
+
+# Copyright 2014-2020 United Kingdom Atomic Energy Authority
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+import os
+import matplotlib.pyplot as plt
+from matplotlib.collections import PolyCollection
+
+from cherab.solps import load_solps_from_raw_output
+
+
+# Load the simulation.
+demos_directory = os.path.dirname(os.path.dirname(__file__))
+simulation_directory = os.path.join(demos_directory, 'data', 'raw')
+print('Loading simulation...')
+sim = load_solps_from_raw_output(simulation_directory)
+
+#get the neccessary data from SOLPSSimulation
+vertices = sim.mesh.vertex_coordinates
+quadrangles = sim.mesh.quadrangles
+mesh_extent = sim.mesh.mesh_extent
+quadrangle_to_grid_map = sim.mesh.quadrangle_to_grid_map
+
+#plot the quadrangle b2 mesh
+collection_mesh = PolyCollection(vertices[quadrangles], facecolor="none")
+
+fig_mesh, ax = plt.subplots(figsize=(10, 20))
+ax.add_collection(collection_mesh)
+ax.set_xlim(mesh_extent["minr"], mesh_extent["maxr"])
+ax.set_ylim(mesh_extent["minz"], mesh_extent["maxz"])
+ax.set_aspect(1)
+ax.set_ylabel("R [m]")
+ax.set_ylabel("Z [m]")
+ax.set_title("Quadrangle B2 Mesh")
+
+#plot the quadrangle b2 mesh with b2 ion temperature values
+collection_values = PolyCollection(vertices[quadrangles])
+collection_values.set_array(sim.ion_temperature[quadrangle_to_grid_map[:,0], quadrangle_to_grid_map[:,1]])
+
+fig_mesh, ax = plt.subplots(figsize=(10, 20))
+ax.add_collection(collection_values)
+ax.set_xlim(mesh_extent["minr"], mesh_extent["maxr"])
+ax.set_ylim(mesh_extent["minz"], mesh_extent["maxz"])
+ax.set_aspect(1)
+ax.set_ylabel("R [m]")
+ax.set_ylabel("Z [m]")
+ax.set_title("B2 Ion Temperature")
```

Comment:
If the `plot_quadrangle_mesh()` is updated as shown above, then this code can be replaced with this:
```python
#plot the quadrangle b2 mesh
ax = sim.mesh.plot_quadrangle_mesh()
ax.set_title("Quadrangle B2 Mesh")
ax.get_figure().set_size_inches((10, 20))

#plot the quadrangle b2 mesh with b2 ion temperature values
ax = sim.mesh.plot_quadrangle_mesh(solps_data=sim.ion_temperature)
ax.get_figure().colorbar(ax.collections[0], aspect=40)
ax.get_figure().set_size_inches((10, 20))
ax.set_title("B2 Ion Temperature [eV]")
plt.show()
```
Also, please add `plt.show()` to the end of all plotting demos so that they can be run correctly without Jupyter.

Reply:
I adjusted the plot demo to show how can the plot methods be used and also left the example of what is done within the methods. I also renamed the file from quadrangle_mesh.py to mesh_and_values.py, since there are now both examples of triangular and quadrangle mesh plots.